SAGE Knowledge.js

```diff
@@ -28,65 +28,65 @@
+		if (getSearchResults(doc)) {
+			return "multiple";
+		}
+	} else {
+		return getItemType(doc, url);
 	}
-	
-	return getItemType(doc);
 }
 
-function getItemType(doc) {
-	var itemType = ZU.xpathText(doc, '//div[@id="contentFrame"]//li[contains(@class, "contentType")]/@class')
-		|| ZU.xpathText(doc, '//div[@id="mainContent"]//p[contains(@class, "docTypeIcon")]/@class');
-	
-	if (!itemType) return;
-	
-	itemType = itemType.replace(/contentType|docTypeIcon/,'').trim();
-	
-	switch(itemType) {
-		case "iconEncyclopedia":
-		case "iconEncyclopedia-chapter":
-			return "encyclopediaArticle";
-		case "iconBook-chapter":
-			return "bookSection";
-		case "iconDictionary-chapter":
-			return "dictionaryEntry";
-		case "iconDebate":
-		case "iconHandbook":
-		case "iconBook":
-		case "iconDictionary":
-		case "iconMajorWork":
-			return "book";
+function getItemType(doc, url) {
+	if (url.split('/').length>5) {
+		//This includes now also encyclopadiaArticles and dictionaryEntries.
+		return "bookSection";
+	} else {
+		return "book";
 	}
 }
 
 function getItem(doc, url) {
-	ZU.doGet(doc.getElementById("_citeLink").href, function(text) {
+	var id = ZU.xpathText(doc, '(//input[@id="contentId"]/@value)[1]');
+	var urlParts = url.split('/');
+	if (urlParts.length>5) {
+		var chapterId = urlParts[urlParts.length-1].replace('.xml', '');
+		var citeLink = '//sk.sagepub.com/CitationExport/ExportEntryCitation/'+id+'?type=Endnote&xmlId='+chapterId;
+	} else  {
+		var citeLink = '//sk.sagepub.com/CitationExport/exportcitation/'+id+'?type=Endnote';
+	}
+	//Z.debug(citeLink)
+
+	ZU.doGet(citeLink, function(text) {
 		var re = /<textarea name="records".*?>([\s\S]*?)<\/textarea>/;
-		var match = re.exec(text)[1]
+		var match = text//re.exec(text)[1]
 			.replace(/NV\s+-\s+1\n/, "")
-			.replace(/^(AU\s+-\s+.+?),? Ph\.? ?D\.?\n/mg, '$1\n');	
+			.replace(/^AU\s+-\s+,\s+$/m, '')
```

Comment:
Yes, but this is an empty author field, i.e. something like
```
AU  -  , 
```
(Note the beginning `^` and end of line `$` characters in the regexp.)